### PR TITLE
[AutoTuner] Add `--seed`

### PR DESCRIPTION
**Pull request by luarss**
Improves reliability of AT experiments/ci 

TODO

- [ ] test_seed()
- [x] Add seed functionality

### CODE REVIEW

File: tools/AutoTuner/src/autotuner/distributed.py [hunk 1]
```diff
@@ -882,18 +896,33 @@ def set_algorithm(experiment_name, config):
...
+    # A wrapper algorithm for limiting the number of concurrent trials.
     if args.algorithm not in ["random", "pbt"]:
         algorithm = ConcurrencyLimiter(algorithm, max_concurrent=args.jobs)
+
+    # Self seed
+    if args.seed is not None:
+        torch.manual_seed(args.seed)
+        np.random.seed(args.seed)
+        random.seed(args.seed)
+
```
Comment (vvbandeira) [inline tools/AutoTuner/src/autotuner/distributed.py]:
```suggestion
```

File: tools/AutoTuner/src/autotuner/distributed.py [hunk 2]
```diff
@@ -870,10 +871,23 @@ def set_algorithm(experiment_name, config):
     """
     Configure search algorithm.
     """
+    # Pre-set seed if user sets seed to 0
+    if args.seed == 0:
+        print("Warning: you have chosen not to set a seed. Do you wish to continue? (y/n)")
+        if input().lower() != "y":
+            sys.exit(0)
+        args.seed = None
```
Comment (vvbandeira) [inline tools/AutoTuner/src/autotuner/distributed.py:885]:
```suggestion
    else:
        torch.manual_seed(args.seed)
        np.random.seed(args.seed)
        random.seed(args.seed)
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

**Review by vvbandeira — APPROVED** [review]

**Comment by vvbandeira** [comment]
FYI @maliberty

**Comment by maliberty** [comment]
Great - will the next step be an ORFS PR to use --seed in the AT test?

**Comment by luarss** [comment]
Seeding will be done by default if `args.seed` is not specified (which is what we do for the rest of the AT tests)

**Pull request merged**